### PR TITLE
[Fix] Cases alpha = 0 or alpha = 1

### LINKED ISSUE #35: Fix the case alpha = 0
When running the dense/sparse solvers with ``alpha = 0``, the following error occurs:
```
  File "/data/parietal/store3/work/pbarbara/fugw/src/fugw/solvers/dense.py", line 425, in solve
    current_loss = compute_fugw_loss(pi, gamma)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/parietal/store3/work/pbarbara/fugw/src/fugw/solvers/dense.py", line 166, in fugw_loss
    "gromov_wasserstein": loss_gromov_wasserstein.item(),
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'item'
```

### PR DESCRIPTION
**Pull request by pbarbarant**
Fixes #35.

TODO list :

- [x] Fix the issue for the dense solver 
- [x] Fix the issue for the sparse solver
- [x] Implement new tests failing under the previous conditions

### CODE REVIEW

**Comment by bthirion** [comment]
It's good practice to add a test corresponding to the configuration that you have addressed.

**Comment by pbarbarant** [comment]
> It's good practice to add a test corresponding to the configuration that you have addressed.

Indeed I added a set of ``alphas = [0.0, 0.5, 1.0]`` to both ``test_dense_solvers`` and ``test_sparse_solvers`` parametrizations to test all possible cases.

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: src/fugw/solvers/dense.py [hunk 1]
```diff
@@ -121,11 +121,11 @@ def fugw_loss(
...
-        loss_gromov_wasserstein = None
-        loss_marginal_constraint_dim1 = None
-        loss_marginal_constraint_dim2 = None
-        loss_regularization = None
+        loss_wasserstein = torch.zeros(1)
+        loss_gromov_wasserstein = torch.zeros(1)
+        loss_marginal_constraint_dim1 = torch.zeros(1)
+        loss_marginal_constraint_dim2 = torch.zeros(1)
+        loss_regularization = torch.zeros(1)
         loss = 0
```
Comment (alexisthual) [inline src/fugw/solvers/dense.py:129]: Out of curiosity, why not simply do `loss_wasserstein = 0`?

File: src/fugw/solvers/sparse.py [hunk 2]
```diff
@@ -167,11 +167,11 @@ def fugw_loss(self, pi, gamma, data_const, tuple_weights, hyperparams):
...
-        loss_gromov_wasserstein = None
-        loss_marginal_constraint_dim1 = None
-        loss_marginal_constraint_dim2 = None
-        loss_regularization = None
+        loss_wasserstein = torch.zeros(1)
+        loss_gromov_wasserstein = torch.zeros(1)
+        loss_marginal_constraint_dim1 = torch.zeros(1)
+        loss_marginal_constraint_dim2 = torch.zeros(1)
+        loss_regularization = torch.zeros(1)
         loss = 0
```
Comment (alexisthual) [inline src/fugw/solvers/sparse.py:175]: Same question here

Comment (pbarbarant) [reply src/fugw/solvers/sparse.py:175]: Same answer as above

File: src/fugw/solvers/dense.py (hunk 1 above)
Comment (pbarbarant) [reply src/fugw/solvers/dense.py:129]: This (I admit counter-intuitive) formulation is compatible with torch's ```.item()``` which previously failed when any loss remained at 0.
```
"wasserstein": loss_wasserstein.item(),
"gromov_wasserstein": loss_gromov_wasserstein.item(),
"marginal_constraint_dim1": loss_marginal_constraint_dim1.item(),
"marginal_constraint_dim2": loss_marginal_constraint_dim2.item(),
"regularization": loss_regularization.item(),
"total": loss.item(),
```

Comment (alexisthual) [reply src/fugw/solvers/dense.py:129]: Thanks for the quick answer!

Comment (pbarbarant) [reply src/fugw/solvers/dense.py:129]: You're welcome ! 😊

**Pull request merged**